### DISCUSSION TITLE
EJBCA CE Docker and REST API Client Authentication

### DISCUSSION THREAD

**Discussion by jarirajari** [Keyfactor/ejbca-ce#715] [Q&A] [answered]
Is anyone able to enable REST API Client Authentication with Dockerized setup. I did install EJBCA CE using docker, and the I created SuperAdmin (cn field) enduser entity. Finally I enabled REST API, for example https://ejbca-node1/ejbca/ejbca-rest-api/v1/ca/status.

The problem is that I cannot get response from this REST API. I tried 1) importing client cert to browser and accessing directly, and 2) using curl by giving URL, client cert .p12, and ca cert. However, curl works with "-k" switch. Before pasting any error messages, it would be great to hear if REST authentication works for anyone. Thanks!

**Comment by primetomas** [comment, marked as answer]
I have done that many times. You have to run the "Quick start - classic workflow" from https://hub.docker.com/r/keyfactor/ejbca-ce. You can not use TLS_SETUP="simple".

**Comment by jarirajari** [comment]
Thanks @primetomas ! I had different entrypoint to documentation and thus used "simple" setup. As per the link you provided got my problems solved. However, accessing REST API via browser produced a bit misleading response since the endpoint was enabled but HTTP Status was 403 when looking it closer. I used curl to access the endpoint successfully. To others, who might see strange problems, I recommend checking the provided link and use the classic workflow as instructed (TLS_SETUP_ENABLED=true).
